### PR TITLE
[MRG+1] Contour levels to colorbar.

### PR DESCRIPTION
**Pull request by jaeilepp**
Closes https://github.com/mne-tools/mne-python/issues/3795

The colorbars are so small by default, that I had to make the tick labels a bit smaller.

### CODE REVIEW

**Comment by jaeilepp** [comment]
colorbar

**Comment by jona-sassenhagen** [comment]
Warning: it is possible, and often desirable, to set 0 contour levels. Wouldn't this then mean you have no units on the colorbar?

Otherwise, you're a beast!

**Comment by jaeilepp** [comment]
Then it only shows the zero, but maybe this should also show ``vmax`` and ``vmin``.

**Comment by jaeilepp** [comment]
Ok. Tried it, and it looks a bit weird, since it's not evenly spaced.
colorbar

**Comment by jaeilepp** [comment]
I'm not sure which is the best way to do this. Opinions?

**Comment by jaeilepp** [comment]
This made me realize that the contour levels aren't actually same for all the images.

**Comment by mmagnuski** [comment]
I also think this shouldn't be the default. For multiple topomaps contour levels would have to be forced to be the same... BTW - how does it look like if you do it the other way: instead of forcing colorbar ticks to be where the contour lines are, force contour lines to be where colorbar ticks have been set?

**Comment by jaeilepp** [comment]
Yeah, that's what I'm looking at now. Trying to find out how mpl does it. I don't want the contours to set to something like 166.6666666.

**Comment by mmagnuski** [comment]
I don't remember ever seeing mpl setting ticks to such weird values. :) But you can access the tick locators in ticker module. I've never played with these classes though...

**Comment by jaeilepp** [comment]
Now it's using the ticker levels and seems to work quite nicely. Here's how it looks with 3 contours.
contours

**Comment by jona-sassenhagen** [comment]
That looks quite good IMO!

A somewhat more far-fetched idea, could similar tech be used to threshold certain t score maps by signifiance levels eventually ..?

**Comment by jaeilepp** [comment]
Probably. I don't know the stats package that well. 

This changes the contour levels a little bit as can be seen from the pictures, but I suppose it's okay since the threshold levels were not previously visible anyway.

**Comment by mmagnuski** [comment]
Looks good!

**Comment by agramfort** [comment]
LGTM

File: mne/viz/topomap.py [hunk 1]
```diff
@@ -1388,6 +1390,12 @@ def plot_evoked_topomap(evoked, times="auto", ch_type=None, layout=None,
     vmax = np.max(vlims)
     cmap = _setup_cmap(cmap, n_axes=len(times), norm=vmin >= 0)
 
+    if isinstance(contours, int):
+        from matplotlib import ticker
+        # nbins = ticks - 1, since 2 of the ticks are vmin and vmax, the
+        # correct number of bins equals to contours + 1.
```
Comment (jona-sassenhagen) [inline mne/viz/topomap.py]: "equals contours + 1" or "is equal to contours +1"

**Comment by agramfort** [comment]
LGTM

can someone else have a look so @jaeilepp can have his +2?

**Comment by jona-sassenhagen** [comment]
I will once I'm within wireless range again.

**Comment by Eric89GXL** [comment]
Sorry to come late, but I actually think that showing the max/min and all the contour levels is cleanest. Even if it's not evenly spaced it's most informative in terms of what everything means in the plot

**Comment by jaeilepp** [comment]
It also looks quite bad in most cases
values
values2
I would omit vmin and vmax.

**Comment by Eric89GXL** [comment]
Yeah that does look bad, and if it's fairly common that it happens then it's not worth it. I suspect it wouldn't be too difficult to come up with a heurestic for when to include hte vmin/vmax based on where the other contours are, but that can be another PR if someone gets bored.

LGTM, @jona-sassenhagen feel free to merge if you're happy

**Comment by jaeilepp** [comment]
Yeah, I would have thought that there would have been an option for this in mpl's locator, but I couldn't find one that works.

**Comment by jona-sassenhagen** [comment]
So we're still set on not having this as a parameter-accessible option? It's gonna become the new default?

**Comment by jaeilepp** [comment]
I'm voting for this as a default. The other options would be more or less arbitrary values.

**Comment by mmagnuski** [comment]
I think this should be accessible through a keyword arg... (`contour_legend`, `contour_colorbar`?)

**Comment by jona-sassenhagen** [comment]
It's not easily user-changeable, isn't it? Which would mean it should be settable with a kwarg.

But I see your point @jaeilepp that the current method is arbitrary and maybe it's not so good to set "arbitrary" as a param.

**Comment by jaeilepp** [comment]
Actually, ``plot_topomap`` now also accepts a list for ``contours``. So maybe we can live without a new kwarg and just update the documentation.

**Comment by mmagnuski** [comment]
I like the fact that `contours` are accepted. I was just a little afraid that if colorbar tick positioning is contour-dependent by default, then colorbar will easily get messy with many contour levels. I mean - it can be the default behavior, but it would be nice if it was possible to turn it off.

**Comment by agramfort** [comment]
what do you suggest? is this good for you to merge?

**Comment by mmagnuski** [comment]
My suggestion (and Jona's) was to add a keyword argument that allows to turn off this behavior, but make it default. @jaeilepp - would that be ok?

**Comment by Eric89GXL** [comment]
@mmagnuski can we add that once we hit a use case that needs it? Or do you already know there will be?

**Comment by mmagnuski** [comment]
@Eric89GXL  sure, I've played with it a bit now and I no longer think that colorbar will get messy for a reasonable number of contour levels. :)

**Comment by agramfort** [comment]
so you give your +1?

**Comment by jona-sassenhagen** [comment]
Well it has my +1 :D

Tried it, looks good.

**Comment by mmagnuski** [comment]
Yes, 👍 from me too :)

**Comment by Eric89GXL** [comment]
Thanks @jaeilepp

**Comment by jona-sassenhagen** [comment]
Thanks @jaeilepp !

**Comment by mmagnuski** [comment]
Thanks @jaeilepp !!

**Pull request merged**